dxw/dxw-cache-control
# Chore/migrate to php8.3

This PR updates the plugin for PHP 8 support. It:

- Bumps the composer platform and required minimum PHP version to 8.2
- Updates Kahlan for PHP8 compatibility
- Runs the CI in PHP 8.2

## How to test

1. Run `composer install && vendor/bin/php-cs-fixer fix --dry-run && vendor/bin/psalm && vendor/bin/kahlan` for the automated tests, which should all run green
2. Spin up a site that uses this plugin and has already been migrated to PHP 8 (https://github.com/dxw/london-travel-watch is a good bet for this)
3. Working locally in that site's repo, update `whippet.json` so it points at this branch of the plugin:
   ```
   {"name": "dxw-cache-control", "src": "git@github.com:dxw/dxw-cache-control.git", "ref": "chore/migrate-to-php8.3"},
   ```
   Then run `whippet deps update plugins/dxw-cache-control`.
4. In the `config/server.php` file, add `define('WP_ENVIRONMENT_TYPE', 'local');`. This allows you to switch on developer mode in the cache control plugin
5. Spin up the site, activate the cache control plugin, and switch on developer mode in its settings page. Set some other settings to whatever values you like, that will be easily viewable in the front-end.
6. Navigate around the front-end of the site, using the console to confirm the developer headings returned to confirm the cache control plugin is still behaving as expected.
